File: app/src/main/java/com/openclassrooms/realestatemanager/data/local/dao/ImageOfPropertyDao.java
package com.openclassrooms.realestatemanager.data.local.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.openclassrooms.realestatemanager.data.local.entities.ImageOfProperty;

import java.util.List;

@Dao
public interface ImageOfPropertyDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long[] insertImagesOfProperty(List<ImageOfProperty> imagesOfProperty);

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    long insertImageOfProperty(ImageOfProperty imageOfProperty);

    @Update
    int updateImageOfProperty(ImageOfProperty imageOfProperty);

    @Query("SELECT * FROM property_image WHERE property_id = :propertyId")
    LiveData<List<ImageOfProperty>> getAllImageForProperty(String propertyId);

    @Query("DELETE FROM property_image WHERE id = :imageId")
    void deleteImage(Integer imageId);
}

File: app/src/main/java/com/openclassrooms/realestatemanager/data/local/dao/ImageOfPropertyDao.kt
package com.openclassrooms.realestatemanager.data.local.dao

import com.openclassrooms.realestatemanager.data.local.entities.ImageOfProperty
import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface ImageOfPropertyDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertImagesOfProperty(imagesOfProperty: List<ImageOfProperty?>?): LongArray?

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insertImageOfProperty(imageOfProperty: ImageOfProperty?): Long

    @Update
    fun updateImageOfProperty(imageOfProperty: ImageOfProperty?): Int

    @Query("SELECT * FROM property_image WHERE property_id = :propertyId")
    fun getAllImageForProperty(propertyId: String?): LiveData<List<ImageOfProperty?>?>?

    @Query("DELETE FROM property_image WHERE id = :imageId")
    fun deleteImage(imageId: Int?)
}